src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ExpressionSyntaxExtensions.cs

```diff
@@ -727,11 +727,12 @@ public static OperatorPrecedence GetOperatorPrecedence(this ExpressionSyntax exp
                 case SyntaxKind.CheckedExpression:
                 case SyntaxKind.UncheckedExpression:
                 case SyntaxKind.AnonymousMethodExpression:
+                case SyntaxKind.SuppressNullableWarningExpression:
```

Comment:
this the meet of the change.  we were not understanding teh precedence of `!` in `x!` and so we were bailing out of removing the parens in `(x!)` later on.  

The rest of the changes are to NRT this code.